### PR TITLE
Use Xcode 15.0.1

### PR DESCRIPTION
**Pull request by jkmassel**
### Migrates the project to Xcode 15 and Apple Silicon in CI

Also includes:
- An adjustment to the `UITextView` subclass that injects a `UIPasteboard` object. This allows us to mock access to the system pasteboard, which requires user interaction in recent iOS versions
- An adjustment to string index handling – in newer versions of Xcode, there are runtime assertions about string index overflow, so we need to be a bit more careful when replacing text 
- A tiny little fix for a failing test, presumably related to something about how iOS generates PNG files.

This can be reviewed by anyone, but I'd appreciate if @mokagio or @crazytonyli took a look at the project code adjustments!

### Testing Details

Ensure CI passes

---

- [x] Please check here if your pull request includes additional test coverage.
- [x] I have considered if this change warrants release notes and have added them to the appropriate section in the `CHANGELOG.md` if necessary.

### CODE REVIEW

File: Aztec/Classes/Extensions/String+RangeConversion.swift
Comment (crazytonyli) [inline Aztec/Classes/Extensions/String+RangeConversion.swift:259]: [The `endIndex` is not an accessible index.](https://developer.apple.com/documentation/swift/string/endindex)

> A string’s “past the end” position—that is, the position one greater than the last valid subscript argument.

**Review by crazytonyli — CHANGES_REQUESTED** [review]

File: Aztec/Classes/Extensions/String+RangeConversion.swift
Comment (AliSoftware) [inline Aztec/Classes/Extensions/String+RangeConversion.swift]:
```suggestion
        (self.startIndex ..< self.endIndex).contains(index)
```

File: .buildkite/pipeline.yml
Comment (crazytonyli) [inline .buildkite/pipeline.yml]: There is a new workaround for this issue: You can pass a `--patch-cocoapods` option to `validate_podspec` and the publish pod utilities. See https://github.com/wordpress-mobile/WordPressKit-iOS/pull/652/files.

File: Aztec/Classes/Extensions/String+RangeConversion.swift
Comment (jkmassel) [reply Aztec/Classes/Extensions/String+RangeConversion.swift:258]: This change causes the following test to fail:

```swift
    func testThatEndIndexIsReportedAsValid() {
        let candidate = "This is a string 👍🏽"
        XCTAssertTrue(candidate.isValidIndex(candidate.endIndex))
    }
```

File: .buildkite/pipeline.yml
Comment (crazytonyli) [inline .buildkite/pipeline.yml]: This `agents` configuration needs to be removed.

**Review by crazytonyli — CHANGES_REQUESTED** [review]
Maybe updating the publishing pod commands in`.buildkite/publish-aztec-pod.sh` too, with the same `--patch-cocoapods` option? Otherwise, future releasing jobs would fail.

File: Aztec/Classes/TextKit/TextView.swift
Comment (crazytonyli) [inline Aztec/Classes/TextKit/TextView.swift:498]: Does that mean this line crashes if the selected attributed string contains an "attribute value" that doesn't conform to `NSCoding`? I don't really know anything about this library's implementation, not sure how likely that's going to happen...

Comment (AliSoftware) [reply Aztec/Classes/TextKit/TextView.swift:498]: That's a good question, but also before the migration to Xcode 15 and newer Swift back when `archivedData` was not marked as `throws`, I'd expect the code would already crash (but maybe with an ObjC NSException rather than a Swift error being thrown)?

That being said, it would be a nice improvement to use `let data = try? …` instead and then make the subsequent call to `storeInPasteboard(encoded: data)` conditional on `if let data`; but not sure if that fix belongs in this PR or separate.

Comment (AliSoftware) [reply Aztec/Classes/TextKit/TextView.swift:498]: Also, looking at the deprecated vs new method signature for `NSKeyedArchiver.archivedData`, I'm wondering if the only case where it could `throw` is if you passed `requiresSecureCoding: true` and the objects conformed to `NSCoding`… but not `NSSecureCoding`.

But since here we use `requiresSecureCoding: false` in the implementation of `archivedData`, maybe that means we'll never throw in practice? Or if we do, that would be under the same conditions as the ones the current version of the library already does, i.e. not related to Xcode 13->Xcode15 migration?

**Review by AliSoftware — APPROVED** [review]

**Comment by AliSoftware** [comment]
Breadcrumbs: I reverted the attribution of the cluster to this repo in Buildkite (https://github.com/Automattic/buildkite-ci/pull/331) — aka reverted this repo to use Intel infra — while this PR is still not merged.

Once we land this PR, don't forget to redo the change in `aztec-editor-ios.tf` to switch it back to use the cluster again.

**Review by crazytonyli — APPROVED** [review]

**Comment by mokagio** [comment]
I wanted to help here and tried to address the unit tests failures via https://github.com/wordpress-mobile/AztecEditor-iOS/pull/1382 but didn't get very far.

**Comment by mokagio** [comment]
Going to merge this despite unit tests failures, to unblock moving to the cluster in CI, which in turns unblocks adopting the latest Xcode, https://github.com/wordpress-mobile/AztecEditor-iOS/pull/1392, which, finally, will unblock https://github.com/wordpress-mobile/AztecEditor-iOS/pull/1391.

I'm still experiencing unit tests issue locally, but I want to see how/if the situation changes once this lands.

**Pull request merged**
